sample updates supporting native_posix_64

pabigot opened the pull request:
The first commit clones the board overrides so Kconfig options are set for this variant.

The second fixes  #17581.  (The cfb-specific patch will be pushed separately to prevent holding this PR up).

pabigot committed:
samples: add board customization for native_posix_64

Sample applications for display functions do not work on native_posix_64
due to missing overrides.  Clone the native_posix_64 Kconfig override
for all samples that have a native_posix Kconfig override.

Signed-off-by: Peter A. Bigot <pab@pabigot.com>

npitre commented:
Also I'd suggest you integrate the use of Z_STRUCT_SECTION_ITERABLE(() 
in the FONT_ENTRY_DEFINE() definition as proposed in #17581.

pabigot committed:
subsys/cfb: correct font definition macro for linker script

This macro provides the required alignment directives to ensure that the
font definitions are placed properly for iteration as members of an
array object.

Closes #17581

Signed-off-by: Peter A. Bigot <pab@pabigot.com>

andyross commented on samples/gui/lvgl/boards/native_posix_64.conf:
```diff
@@ -0,0 +1,5 @@
+CONFIG_SDL_DISPLAY=y
+CONFIG_SDL_DISPLAY_DEV_NAME="DISPLAY"
+CONFIG_SDL_DISPLAY_X_RES=320
+CONFIG_SDL_DISPLAY_Y_RES=240
```
I don't claim to understand the nature of the problem here, but if we're having to do the same hack in two otherwise unrelated samples for two distinct platforms, it's probably time to come up with a framework to manage this better.  That is, the native_posix platforms would declare CONFIG_NEED_YADAYADA_SDL_WORKAROUND and the SDL code would implement the DEV_NAME and RES value fallbacks so the samples and tests don't have to.

vanwinkeljan replied:
@andyross actually `CONFIG_SDL_DISPLAY` is not need as it is set by default for native_posix targets if `CONFIG_DISPLAY` is enabled and in this particular case the resolution parameters are also not needed as they are the default values. So this file could be reduced to `CONFIG_SDL_DISPLAY_DEV_NAME="DISPLAY"`, we could opt to use `DISPLAY` as default value.

andyross reviewed (approved):
Nitpicking on the SDL stuff, about which I know nothing.  But once the same boilerplate gets pasted into four different spots, it's probably time to refactor.

jfischer-no reviewed (approved):
LGTM
